### DISCUSSION TITLE
Welcome to astral Discussions!

### DISCUSSION THREAD

**Discussion by sj14** [sj14/astral#4] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by user197283** [comment]
Hello (python newbie)! I was wondering if it is possible to gather sunset values for more than one date? I noticed there is a singular date parameter, but I am looking for an output of sunset times for several years. Eventually, have a data frame with column1 to be the dates I am interested in, and column2 to have the sunset times for that day. Is that possible? Thanks :)

**Reply by sj14** [reply]
Hi @user197283, 
as you mention Python, you are probably using this astral library: https://github.com/sffjunkie/astral/.
However, when I understand your question correctly, you should be able to loop over an array with the given dates and call the sunset calculation for each date inside the loop, e.g.:

```python
import astral
from astral import sun
from datetime import datetime
import pytz

obs = astral.Observer(0.0, 0.0, 0.0)

for day in range(1, 15):
    date = datetime(2021, 6, day, tzinfo=pytz.utc)
    sunset = sun.sunset(obs, date)
    print("Sunset on {0} was at {1}".format(date.date(), sunset.time()))
```

```text
$ python3.9 ./test.py          
Sunset on 2021-06-01 was at 18:01:20.255373
Sunset on 2021-06-02 was at 18:01:30.058958
Sunset on 2021-06-03 was at 18:01:40.220011
Sunset on 2021-06-04 was at 18:01:50.720402
Sunset on 2021-06-05 was at 18:02:01.541408
Sunset on 2021-06-06 was at 18:02:12.663739
Sunset on 2021-06-07 was at 18:02:24.067571
Sunset on 2021-06-08 was at 18:02:35.732572
Sunset on 2021-06-09 was at 18:02:47.637938
Sunset on 2021-06-10 was at 18:02:59.762424
Sunset on 2021-06-11 was at 18:03:12.084379
Sunset on 2021-06-12 was at 18:03:24.581782
Sunset on 2021-06-13 was at 18:03:37.232275
Sunset on 2021-06-14 was at 18:03:50.013207
```

**Reply by user197283** [reply]
Thank you soooo much, this is exactly what I was looking for. I appreciate the help! :)
